chips/nrf51/src/radio.rs

```diff
@@ -300,3 +283,17 @@ impl nrf5x::ble_advertising_hil::BleAdvertisementDriver for Radio {
         self.tx_client.set(Some(client));
     }
 }
+
+// The capsule validates that the `tx_power` is between -20 to 10 dBm but then
+// chip must validate if the current `tx_power` is supported as well
+impl nrf5x::ble_advertising_hil::BleConfig for Radio {
+    fn set_tx_power(&self, power: u8) -> kernel::ReturnCode {
+        match nrf5x::constants::TxPower::from_u8(power) {
+            TxPower::Error => kernel::ReturnCode::ENOSUPPORT,
+            e @ _ => {
```

Comment:
Sure, but I'm more in favor self-documenting code though!
What about writing out the patterns explicitl instead?
For example as:

```rust
        match nrf5x::constants::TxPower::from_u8(power) {
            // Invalid transmitting power, propogate error
            TxPower::Error => kernel::ReturnCode::ENOSUPPORT,
            // Valid transmitting power, propogate success
            // e - enum variant of nrf5x::constants::TxPower that defines valid transmitting power
            e @ TxPower::Positive4dBM |
            e @ TxPower::Positive3dBM |
            e @ TxPower::ZerodBm |
            e @ TxPower::Negative4dBm |
            e @ TxPower::Negative8dBm |
            e @ TxPower::Negative12dBm |
            e @ TxPower::Negative16dBm |
            e @ TxPower::Negative20dBm |
            e @ TxPower::Negative40dBm => {
                self.tx_power.set(e);
                kernel::ReturnCode::SUCCESS
            }
        }
```

Reply:
Would it make more sense to just use the [`TryFrom`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) trait? It returns a `Result`, which would lead to clearer match statement:

```rust
match TxPower::try_from(power) {
    Err(_) => kernel::ReturnCode::ENOSUPPORT,
    Ok(res) => {
        self.tx_power.set(res);
        kernel::ReturnCode::SUCCESS
    }
}
```

Reply:
Yeah, that is very clean and elegant but still explicit patterns is useful for readability IMO.

@brghena @alevy what do you prefer?

Reply:
I would personally use the TryFrom method Amit proposed (now that I know it exists). But the difference between the two is pretty negligible and I think either is fine.